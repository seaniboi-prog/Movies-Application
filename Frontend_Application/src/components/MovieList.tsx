import React from 'react'
import { Movie } from '../models';
import MovieThumbnail from './MovieThumbnail';
import './mainPage.scss';

interface Props {
    movies: Movie[];
}

const MovieList:React.FC<Props> = ({movies}: Props) => {
    return (
    <div className='movie__list' data-testid="movieList">
        { movies.map(movie => <MovieThumbnail key={movie.id} movie={movie}/>) }
    </div>
    );
}

export default MovieList