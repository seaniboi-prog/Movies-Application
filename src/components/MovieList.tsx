import React from 'react'
import { Movie } from '../models';
import MovieThumbnail from './MovieThumbnail';
import './styles.scss'

interface Props {
    movies: Movie[];
}

const MovieList:React.FC<Props> = ({movies}: Props) => {
    return (
    <div>
        { movies.map(movie => <MovieThumbnail movie={movie}/>) }
    </div>
    );
}

export default MovieList