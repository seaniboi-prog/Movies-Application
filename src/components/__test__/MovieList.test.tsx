import React from 'react'
import ReactDOM from 'react-dom'
import MovieList from '../MovieList'

import { render, cleanup } from '@testing-library/react'
import 'jest-dom/extend-expect'
import renderer from 'react-test-renderer'