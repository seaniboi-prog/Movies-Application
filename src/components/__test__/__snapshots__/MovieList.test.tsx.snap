// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie List Component Tests Matches Snapshot 1`] = `
<div
  className="movie__list"
  data-testid="movieList"
>
  <div
    className="movie__thumbnail"
    data-testid="movieThumbnail"
  >
    <div
      className="movie__box"
      data-testid="movieBox"
      onClick={[Function]}
    >
      <img
        alt="key1"
        className="movie__image"
        data-testid="movieImage"
        src="movie_images/img1.jpg"
      />
      <p
        className="movie__title"
        data-testid="movieTitle"
      >
        name1
      </p>
    </div>
  </div>
  <div
    className="movie__thumbnail"
    data-testid="movieThumbnail"
  >
    <div
      className="movie__box"
      data-testid="movieBox"
      onClick={[Function]}
    >
      <img
        alt="key2"
        className="movie__image"
        data-testid="movieImage"
        src="movie_images/img2.jpg"
      />
      <p
        className="movie__title"
        data-testid="movieTitle"
      >
        name2
      </p>
    </div>
  </div>
</div>
`;
